src/Bicep.Core/Registry/TemplateSpecRepositoryFactory.cs

```diff
@@ -4,20 +4,23 @@
 using System;
 using Azure.Core;
 using Azure.Identity;
-using Azure.ResourceManager.Resources;
+using Azure.ResourceManager;
 
 namespace Bicep.Core.Registry
 {
     public class TemplateSpecRepositoryFactory : ITemplateSpecRepositoryFactory
     {
-        public ITemplateSpecRepository CreateRepository(Uri? endpoint, string subscriptionId, TokenCredential? tokenCredential = null)
+        public ITemplateSpecRepository CreateRepository(Uri? endpointUri, string subscriptionId, TokenCredential? tokenCredential = null)
```

Comment:
>TokenCredential [](http://example.com/codeflow?start=97&length=15)

Btw, does this actually need to be nullable?

Reply:
This is something I'd like to discuss with the team. Basically `endpointUri` is only needed if the user need to deploy to other environments like DogFood or national clouds. With the current implementation the endpoint URI is an optional part of the template reference string, but I am thinking about moving it to `bicepconfig.json` so that users can reuse the the same bicep file for different cloud environments (say they can also set aliases for subscription IDs, resource group name, and template spec names).